ui/admin/app/components/form/policy/index.js

```diff
@@ -0,0 +1,172 @@
+
+export default class FormPolicyComponent extends Component {
+  //attributes
+
+  @tracked is_custom_retention_selected = this.args.model.retain_for?.days > 0;
+  @tracked is_custom_deletion_selected = this.args.model.delete_after?.days > 0;
+  @tracked retain_for_overridable = this.args.model.retain_for?.overridable;
+  @tracked delete_after_overridable = this.args.model.delete_after?.overridable;
+
+  @tracked delete_days = this.args.model.delete_after?.days;
+  @tracked retain_days = this.args.model.retain_for?.days;
+
+  //methods
+
+  /**
+   * Returns retention policy options list
+   * @type {array}
+   */
+  get listRententionOptions() {
+    return RETENTION_POLICY;
+  }
+
+  /**
+   * Returns deletion policy options list, if the retain days are -1 (forever)
+   * then we should force the delete option to be do not delete
+   * @type {array}
+   */
+  get listDeletionOptions() {
+    if (this.retain_days < 0) {
+      return { do_not_delete: 0 };
+    } else {
+      return DELETION_POLICY;
+    }
+  }
+
+  /**
+   * Force the delete After select list to have `do not delete` option selected
+   * @type {boolean}
+   */
+  get disableDeleteOptions() {
+    return this.retain_days < 0 ? true : false;
+  }
+
+  /**
+   * Returns policy type
+   * @type {string}
+   */
+  get selectRetentionPolicyType() {
+    if (this.retain_days < 0) {
+      return 'forever';
+    } else if (this.retain_days >= 1) {
+      return 'custom';
+    } else {
+      return 'do_not_protect';
+    }
+  }
+
+  /**
+   * Returns policy type
+   * @type {string}
+   */
+  get selectDeletePolicyType() {
+    return this.delete_days > 0 && this.is_custom_deletion_selected
+      ? 'custom'
+      : 'do_not_delete';
+  }
+
+  //actions
+  /**
+   * Show custom text field when custom option is selected
+   */
+
+  @action
+  handlePolicyTypeSelection({ target: { value: selectedVal, name: policy } }) {
+    switch (policy) {
+      case 'retention_policy':
+        this.retain_days = selectedVal;
+        //When `forever` is selected, we should automatically select `do_not_delete`
+        //and hide the custom input
+        if (selectedVal < 0) {
+          this.delete_days = 0;
+          this.is_custom_deletion_selected = false;
+        }
+        if (selectedVal > 0) {
+          this.is_custom_retention_selected = true;
+        } else {
+          this.is_custom_retention_selected = false;
+        }
+        break;
+
+      case 'deletion_policy':
+        this.delete_days = selectedVal;
+        if (selectedVal > 0) {
+          this.is_custom_deletion_selected = true;
+        } else {
+          this.is_custom_deletion_selected = false;
+        }
+        break;
+    }
+  }
+
+  /**
+   * Toggle retain for & delete after overridables
+   */
+  @action
+  handleOverridableToggle(field) {
+    this[field] = !this[field];
+  }
+
+  /**
+   * Set retain for input value to a tracked property
+   */
+  @action
+  handleRetainForInput({ target: { value } }) {
```

Comment:
Aren't these the same handlers in essence, couldn't we just take in a variable that says which value we should update?